src/main/java/nextstep/subway/path/domain/PathFinder.java

```diff
@@ -0,0 +1,75 @@
+package nextstep.subway.path.domain;
+
+import nextstep.subway.constant.ErrorMessage;
+import nextstep.subway.exception.BadRequestException;
+import nextstep.subway.exception.NotFoundException;
+import nextstep.subway.line.domain.Line;
+import nextstep.subway.path.dto.ShortestPathResponse;
+import nextstep.subway.path.dto.StationsResponse;
+import nextstep.subway.station.domain.Station;
+import nextstep.subway.station.domain.Stations;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class PathFinder {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PathFinder.class);
+
+    public static ShortestPathResponse findShortestPath(List<Line> allLines, Station starting, Station destination) {
+        validateSameStation(starting, destination);
+
+        WeightedMultigraph<Station, DefaultWeightedEdge> stationGraph = createStationGraph(allLines);
+        validateIsExistStation(stationGraph, starting, destination);
+
+        DijkstraShortestPath<Station, DefaultWeightedEdge> shortestPath = new DijkstraShortestPath<>(stationGraph);
+        GraphPath<Station, DefaultWeightedEdge> path = shortestPath.getPath(starting, destination);
+        validateConnectedStation(path);
+
+        List<Station> stations = path.getVertexList();
+        double distance = shortestPath.getPathWeight(starting, destination);
+
+        return new ShortestPathResponse(StationsResponse.of(new Stations(stations)), (int) distance);
+    }
+
+    private static WeightedMultigraph<Station, DefaultWeightedEdge> createStationGraph(List<Line> lines) {
+        WeightedMultigraph<Station, DefaultWeightedEdge> stationGraph = new WeightedMultigraph(DefaultWeightedEdge.class);
+
+        lines.forEach(
+                line -> line.getSections()
+                        .toList()
+                        .forEach(section -> {
+                            stationGraph.addVertex(section.getUpStation());
+                            stationGraph.addVertex(section.getDownStation());
+                            stationGraph.setEdgeWeight(stationGraph.addEdge(section.getUpStation(), section.getDownStation()), section.getDistance().toInt());
+                        })
+        );
```

Comment:
```suggestion
       lines.stream()
            .flatMap(line -> line.getSectionCollection().stream())
            .forEach(section -> addToGraph(section));
```

이런 식으로 좀 더 짧고 간결한 메서드를 작성해 볼 수 도 있을 것 같아요 :)